server/conf.go

```diff
@@ -588,7 +588,9 @@ func ConfigureOptions(fs *flag.FlagSet, args []string, printVersion, printHelp,
 	if clusterPeers != "" {
 		sopts.Clustering.Peers = []string{}
 		for _, p := range strings.Split(clusterPeers, ",") {
-			sopts.Clustering.Peers = append(sopts.Clustering.Peers, strings.TrimSpace(p))
+			if p = strings.TrimSpace(p); p != sopts.Clustering.NodeID {
```

Comment:
Doing the work here means that it will work in case one start the streaming server from executable, not programmatically. So I wonder if the processing should not simply be done in `RunServerWithOpts()`. But we could first go with your approach and see if it becomes an issue, move that into the run method.